### PR TITLE
SuperTuxKart Game Server that allows AI connections

### PR DESCRIPTION
**Pull request by markmandel**
To allow connections from AI bots from any host required a change to the
source code of Super Tux Kart.

This uses the author's fork to allow the connections, and builds an
image from there.

Increments the Super Tux Kart image as well.

### CODE REVIEW

File: examples/supertuxkart/main.go [hunk 1]
```diff
@@ -28,10 +28,10 @@ import (
 	sdk "agones.dev/agones/sdks/go"
 )
 
-// logLocation is the path to the location of the SuperTuxKart log file
+// logLocation is the path to the location of the Super Tux Kart log file
```
[the remark this answers is not in the document]
Comment (markmandel) [reply examples/supertuxkart/main.go]: Aaah. Somehow that got into the thing. I will revert!

File: examples/supertuxkart/Dockerfile [hunk 2]
```diff
@@ -12,36 +12,54 @@
...
+FROM debian:buster as stk-builder
+
+RUN apt update && apt-get install -y build-essential cmake libbluetooth-dev \
+libcurl4-openssl-dev libenet-dev libfreetype6-dev libharfbuzz-dev libfribidi-dev \
+libgl1-mesa-dev libglew-dev libjpeg-dev libogg-dev libopenal-dev libpng-dev \
+libssl-dev libvorbis-dev libxrandr-dev libx11-dev nettle-dev pkg-config zlib1g-dev \
+git subversion
+
+RUN git clone -b allow-ai https://github.com/markmandel/stk-code.git
+RUN svn co https://svn.code.sf.net/p/supertuxkart/code/stk-assets stk-assets
```
[the remark this answers is not in the document]
Comment (markmandel) [reply examples/supertuxkart/Dockerfile]: Yeah, I had no issue at all. Weird!

File: examples/supertuxkart/Dockerfile [hunk 3]
```diff
@@ -12,36 +12,54 @@
...
+RUN rm -rf ./stk-code/.git
+RUN rm -rf ./stk-assets/.svn
+
+RUN cd stk-code && mkdir cmake_build && cd cmake_build && cmake .. -DBUILD_RECORDER=off \
```
[the remark this answers is not in the document]
Comment (markmandel) [reply examples/supertuxkart/Dockerfile]: Ooh, lemme take that for a spin.

[the remark this answers is not in the document]
Comment (markmandel) [reply examples/supertuxkart/Dockerfile]: Shaves 200M off!

File: examples/supertuxkart/Dockerfile [hunk 4]
```diff
@@ -12,36 +12,55 @@
...
+RUN apt update && apt-get install -y build-essential cmake libbluetooth-dev \
+libcurl4-openssl-dev libenet-dev libfreetype6-dev libharfbuzz-dev libfribidi-dev \
+libgl1-mesa-dev libglew-dev libjpeg-dev libogg-dev libopenal-dev libpng-dev \
+libssl-dev libvorbis-dev libxrandr-dev libx11-dev nettle-dev pkg-config zlib1g-dev \
+git subversion
+
+RUN git clone -b allow-ai https://github.com/markmandel/stk-code.git
+RUN svn co https://svn.code.sf.net/p/supertuxkart/code/stk-assets stk-assets
+RUN rm -rf ./stk-code/.git
+RUN rm -rf ./stk-assets/.svn
```
Comment (roberthbailey) [inline examples/supertuxkart/Dockerfile]: Sorry I didn't catch this the first time. If you don't want the source control bits, why not `svn export`? It looks like you can [even do it for git repositories](https://stackoverflow.com/questions/160608/do-a-git-export-like-svn-export) (that are on github)!

Comment (markmandel) [reply examples/supertuxkart/Dockerfile]: Ooh. Good idea. Let me take this for a spin.

At the very least, I can do a --depth on the git checkout.

**Pull request merged**